internal/svcmanager/svcmanager.go

```diff
@@ -58,9 +65,14 @@ func (m *Manager) WaitWithContext(ctx context.Context) error {
 		case <-ctx.Done():
 			return nil
 		default:
-			if m.Ready() {
+			err := m.Ready()
+			if err == nil {
 				return nil
 			}
+			if m.checkVersion && errs.Matches(err, errVersionMismatch{}) {
```

Comment:
Checking `m.checkVersion` seems unnecessary. Or said differently; if `m.checkVersion=false` then `errVersionMismatch` should never happen in the first place.